<?xml version="1.0"?>
<launch>
  <!-- Listen to messages from joysicks -->
  <node pkg="joy" name="joy_node" type="joy_node"/>

  <!-- Safety Copilot Node -->
  <!--node pkg="racecar_simulator" name="safety_copilot" type="safety_copilot" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node-->


  <!-- Launch a map from the maps folder-->
  <arg name="map" default="$(find racecar_simulator)/maps/levine.yaml"/>
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map)"/>

  <!-- Launch the racecar model -->
  <include file="$(find racecar_simulator)/launch/racecar_model.launch"/>

  <!-- Launch scan converter -->
  <!--rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  <node pkg="racecar_simulator" name="scan_converter" type="scan_converter" output="screen">
  </node -->

  <!-- Begin the simulator with the parameters from params.yaml -->
  <node pkg="racecar_simulator" name="racecar_simulator" type="simulator" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node>

  <!-- Launch the mux node with the parameters from params.yaml -->
  <node pkg="racecar_simulator" name="mux_controller" type="mux" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node>

  <!-- Launch the behavior controller node with the parameters from params.yaml -->
  <node pkg="racecar_simulator" name="behavior_controller" type="behavior_controller" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node>

  <!-- Launch the Random Walker Node -->
  <node pkg="racecar_simulator" name="random_walker" type="random_walk" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node>

  <!-- Launch the Keyboard Node -->
  <node pkg="racecar_simulator" name="keyboard" type="keyboard" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node>

  <!-- ***Put launch command for new planner here:-->
  <!-- Launch the New Planner Node -->
  <!--   <node pkg="racecar_simulator" name="new node's name" type="new file name" output="screen">
          <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  	 </node>    -->

  
  <!-- Launch RVIZ -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find racecar_simulator)/launch/new_simulator.rviz" output="screen"/>
</launch>
